test/functional/user_sessions_controller_test.rb

```diff
@@ -170,6 +170,22 @@ class UserSessionsControllerTest < ActionController::TestCase
       post :create
       assert_equal "Successfully logged in.",  flash[:notice]
     end
+  
+    test 'sign up and login via provider basic flow for twitter user with no email' do
+      assert_not_nil OmniAuth.config.mock_auth[:twitter_no_email]
+      #Omniauth hash is present
+      request.env['omniauth.auth'] =  OmniAuth.config.mock_auth[:twitter_no_email]
+      assert_not_nil request.env['omniauth.auth']
+      #Sign Up for a new user
+      post :create
+      assert_equal "You have successfully signed in. Please change your password by editing your profile in the upper right menu.",  flash[:notice]
```

Comment:
OK, we'll have to change this too, from `assert_equal "You have successfully signed in. Please change your password using the link sent to you via e-mail.",  flash[:notice]`

Reply:
https://github.com/publiclab/plots2/blob/4b2e5321de6cc0c2341535b722722c1bfa1804f0/app/controllers/user_sessions_controller.rb#L92-L94